File: src/main/java/com/vaadin/demo/component/popover/PopoverModal.java
package com.vaadin.demo.component.popover;

import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.popover.Popover;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("popover-modal")
public class PopoverModal extends Div {

    public PopoverModal() {
        // tag::snippet[]
        Popover popover = new Popover();
        popover.setModal(true);
        popover.setBackdropVisible(true);
        // end::snippet[]
        TextField code = new TextField("Discount code");
        Button apply = new Button("Apply");
        popover.add(code, apply);

        Button button = new Button("Discount");
        popover.setTarget(button);

        add(button, popover);
    }

    public static class Exporter extends DemoExporter<PopoverModal> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/popover/popover-modal.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/button';
import '@vaadin/popover';
import '@vaadin/text-field';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('popover-modal')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-button id="target">Discount</vaadin-button>
      <!-- tag::snippet[] -->
      <vaadin-popover for="target" modal with-backdrop>
        <vaadin-text-field label="Discount code"></vaadin-text-field>
        <vaadin-button>Apply</vaadin-button>
      </vaadin-popover>
      <!-- end::snippet[] -->
    `;
  }
}
